content/blog/sivayanama-simplifying-open-api-specification-with-reusable-common-data-models.md

```diff
@@ -0,0 +1,114 @@
+Components object in the OpenAPI specification is the home of reusable object definitions. However, these defined objects must be explicitly referred to outside of the components section wherever required.
+
+```yaml
+components:
+  parameters:
+    tenantId:
+      in: path
+      name: tenantId
+      schema:
+        type: string
+      required: true
+      description: Describes the clientId or mspId of tenant
+```
+
+*$ref* is one of the fixed fields in the schema. It is a string value which refers to other components in the OpenAPI document, internally and externally. The above defined *tenantId* parameter can be referred to below: 
+
+
+```yaml
+$ref: '#/components/parameters/tenantId’
+```
+
+## Externalized definition
+Data models can be defined outside of the OpenAPI specification file using a $ref reference to an external definition. For example:
+
+```yaml
+example-multiple-threshold-type-example-request:
+   $ref: ./models/opsramp-monitoring-management/multiple-threshold-type-example-request-v1.yaml
+```
+### Content of multiple-threshold-type-example-request-v1.yaml
+Please note that the request object definition should be defined within the *value:* as shown below.
+
+```yaml
+value:
+  id: 1ecf993f-9b54-4ce3-9581-c365188f7e58
+  name: OpsRamp Gateway Performance Template
+  description: Monitors basic UNIX parameters like UNIXCPU, UNIXSTORAGE, UNIXUPTIME,
+    UNIXMEMORY, UNIXLOAD and UNIXStats
+  resourceType: DEVICE
+  collectorType: OpsRamp Gateway
+  status:
+    state: ACTIVE
+  generation: 2
+  tags: Performance Monitors
+  createdDate: '2022-10-09T15:03:44+0000'
+  updatedDate: '2022-10-09T15:23:40+0000'
+  scope: Client
+  templateFamily: Performance Monitors Family
+  notes: Sample notes related to performance monitors
+```
+
+## Externalized definition with local aliases 
+The above externalized definition can be further improved by defining local aliases. The local aliases can be used instead of repeating the relative path of the definition in all references. In the example shown below, the external definition referenced with the $ref can be referred by *#components/parameters/tenanId*:
+
+```yaml
+components:
+  parameters:
+    tenantId:
+      $ref: ./models/opsramp-monitoring-management/tenantId-v1.yaml
+      # this can be referred by #components/parameters/tenantId
+```
+
+## Advantages of externalised definitions 
+Externalized definitions have many advantages over traditional inline definitions, such as:
+
+- Loosely coupling data models, and definitions from OpenAPI Specification.
+- Data models can be reused with common definitions.
+- This will reduce the OpenAPI specification files significantly.
+- Easy to manage and govern OpenAPI specification files.
+   
+
+## Conclusion
+In this article, I have explained the four techniques to simplify the OpenAPI specification using loosely coupled, reusable data model definitions. I have also explained the advantages of these approaches. 
```

Comment:
Consider modifying to:
In this article, I explained four techniques used to simplify the OpenAPI Specification using loosely coupled, reusable data model definitions and pointed out the advantages of these approaches. Check back for more articles on this and other subjects on the HPE Developer blog. (and then hyperlink to this URL: https://developer.hpe.com/blog )